introduction-to-qgis.Rmd

```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_08.png')
 ```
 
-9. 
+
+9. Again right-click on ``banglore_lakes`` layer and click *Properties* to open the properties dialog. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_09.png')
 ```
-10. 
+
+10. In the *Layer Properties* dialog, choose *Attribute Form*. Under *Fields*, select *status*. Choose the *Widget Type* as ``Value Map``. Value Map allows us to create a drop-down while adding field values for each feature(polygon).
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_10.png')
 ```
 
-11. 
+
+11. Enter *Value* as ``1``, *Description* as ``Healthy``. *Value* as ``2``, *Description* as ``Partially Lost``, and *Value* as ``3``, *Description* as ``Lost``. The status field will be set as a drop-down with these three values.
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_11.png')
 ```
 
-12. 
+12. We require this field's input must be given for every feature, so let's make this a mandatory field. Under *Constraints* check *Not null* and *Enforce not null constraint*. Click *Ok* to save the changes and close the properties dialog. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_12.png')
 ```
 
-13. 
+13. In the QGIS main canvas, before editing, let us enable the snapping toolbar. This toolbar will help select the nearby vertex corners and avoid invalid geometries. Right-click on the toolbar and check the *Snapping Toolbar* to enable it.
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_13.png')
 ```
 
-14. 
+14. Now the snapping toolbar will be added to the main QGIS.  
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_14.png')
 ```
 
-15. 
+15. To start digitizing, select the ``banglore_lakes`` layer. Click the *Toggle editing* and the *Add Polygon Feature* button. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_15.png')
 ```
 
-16. 
+16. Enable snap setting by clicking *Enable Snapping* button. Zoom to any water body region. Starting from an edge, keep on adding vertices using left-click. Once the polygon is fully covered, right-click to complete the feature digitization. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_16.png')
 ```
 
-17. 
+17. Now the *Feature Attributes* dialog will popup to get the attributes value. By referring to the ``OSM Standard`` basemap, enter the *name* and select the *status* of the waterbody. Click *Ok*. 
```

Comment:
Now the *Feature Attributes* dialog will popup to get the attributes value -> Now you will be prompted by the *Feature Attributes* dialog to enteer the attributes values for the feature you just created.

Reply:
Updated.